### ISSUE TITLE
Gain

### ISSUE THREAD

**Issue by mzed** [immersive-audio-live/ADM-OSC#9]
We currently have linear gain messages in 0.4.  Some of the new messages proposed for 0.5 call for gain in dB.  This is potentially confusing.

Maybe those messages should be `/gaindB`?

**Comment by mzed** [comment]
`/gaindB` makes sense in relation to ADM, since the gain element has an attribute "gainUnits" to set the units to dB. The default is linear.

**Comment by mzed** [comment]
/channel/1/gain -60

or 

/channel/gain 1 -60

———
/channel/*/gain -60

**Comment by mzed** [comment]
Lawo: Normalized gain solves min & max; second message creates more opportunity for failure

**Comment by Daniel-Higgott** [comment]
Is there a standardisation for how the normalised gain maps to the gain in dB though? If a sender wants to set an object to a specific value in dB, or indeed adjust by an amount in dB, how would that sender know how to achieve this?

If the gain was in dB could it not be up to the receiver to set min/max thresholds?

**Comment by madees** [comment]
![image](https://github.com/immersive-audio-live/ADM-OSC/assets/67159179/b46c2402-91c5-4a3b-b655-3bfe63c09e79)

issue with /gaindB is the [-INF] value to transmit as a float. Trade of could be to define a corresponding very low value like -120.0 dB

**Comment by Daniel-Higgott** [comment]
This works if your gain range is between -inf and 0dB. However, if an application wanted to apply positive gain, the linear representation would be above 1.0. Currently the ADM-OSC specification is expecting a value between 0.0 and 1.0.

**Comment by mzed** [comment]
Lawo's argument about min and max (above) is that /gain 0. represents the system's minimum gain and /gain 1. represents maximum gain.  So, if your gain slider goes from -80db to +12db, then that's what 0.0-1.0 will get.

ADM does define a gain of 1.0 as [no change](https://adm.ebu.io/reference/adm_elements/parameters/gain.html).  This is the default value for gain. Gain is just a multiplier, so to get +12db the gain would be 4.0.

Either way we have some complexity. If gain 1.0 is "top of the fader" then we don't really know what gain we're getting. But if gain 1.0 == 0 dB, then we don't know if higher gain is possible.  I would tend to stick with ADM here. There's no reason ADM-OSC can't support gain > 1.0, but a specific renderer might not support that. (L-ISA doesn't.)

**Comment by Daniel-Higgott** [comment]
@mzed - I think the solution is that the equations in @madees answer above should be the standard for converting between linear and logarithmic gain. The equations should probably be referenced directly in the ADM OSC specification.

In the specification the 'max' value shouldn't be 1.0 (as it currently is), but could be either higher or undefined (?). What is defined by those equations is that a value of 1.0 is 0dB (or unchanged) and 0.0 is -inf. It's up to the client to define the maximum value it accepts.

**Comment by mzed** [comment]
Yes, this makes sense to me.  ADM-OSC supports linear gain as positive floats. Gain dB can be derived with the above equations.

Maybe we need a /maxGain query?
